server/datastore/mysql/android.go

```diff
@@ -1632,6 +1632,37 @@ WHERE
 	})
 }
 
+// HasAndroidAppConfigurationChanged checks if the new configuration for an Android app
+// identified by adam_id and global_or_team_id is different from the existing one. This
+// is a datastore method so that we rely on mysql's canonicalisation of JSON for comparison.
+func (ds *Datastore) HasAndroidAppConfigurationChanged(ctx context.Context, adamID string, globalOrTeamID uint, newConfig json.RawMessage) (bool, error) {
```

Comment:
nit: maybe `applicationID`? I think `adamID` is a very Apple/VPP app concept and ideally we'll rename the DB cols at some point 😅